core/src/main/java/com/xmartlabs/base/core/module/GsonModule.java

```diff
@@ -22,14 +23,19 @@
 
 import dagger.Module;
 import dagger.Provides;
+import lombok.Setter;
 
 @Module
 public class GsonModule {
   static final String SERVICE_GSON_NAME = "ServiceGson";
 
+  @Setter
+  @Nullable
+  private Function<GsonBuilder, GsonBuilder> modifyGsonBuilder;
```

Comment:
Did you think about using inheritance in order to avoid using it? Maybe we need different  modifiers to the service gson than the common modifiers, I don't know.